src/lib/operators/table_scan.cpp

```diff
@@ -0,0 +1,167 @@
+#include "table_scan.hpp"
+#include "all_type_variant.hpp"
+#include "resolve_type.hpp"
+#include "storage/abstract_attribute_vector.hpp"
+#include "storage/dictionary_segment.hpp"
+#include "storage/reference_segment.hpp"
+#include "storage/table.hpp"
+#include "storage/value_segment.hpp"
+
+namespace opossum {
+
+TableScan::TableScan(const std::shared_ptr<const AbstractOperator>& in, const ColumnID column_id,
+                     const ScanType scan_type, const AllTypeVariant search_value) {
+  _in = in;
+  _column_id = column_id;
+  _scan_type = scan_type;
+  _search_value = search_value;
+}
```

Comment:
```suggestion
TableScan::TableScan(const std::shared_ptr<const AbstractOperator>& in, const ColumnID column_id,
                     const ScanType scan_type, const AllTypeVariant search_value)
                     :  AbstractOperator(in), _column_id(column_id), _scan_type(scan_type), _search_value(search_value) {}
```
By calling `AbstractOperator(in)` you can now also use `_left_input_table()` instead of `_in->get_output()`.